unblob/handlers/archive/cpio.py

```diff
@@ -93,7 +93,34 @@ class BinaryHandler(_CPIOHandlerBase):
             $cpio_binary_magic= { c7 71 } // (default, bin, hpbin)
 
         condition:
-            $cpio_binary_magic
+            $cpio_binary_magic and
+            /**
+                check that c_mode file type is valid
+                C_ISBLK		060000 (0x6)
+                C_ISCHR		020000 (0x2)
+                C_ISDIR		040000 (0x4)
+                C_ISFIFO	010000 (0x1)
+                C_ISSOCK	0140000 (0xc)
+                C_ISLNK		0120000 (0xa)
+                C_ISCTG		0110000 (0x9)
+                C_ISREG		0100000 (0x8)
+            */
+            for any i in (0x1, 0x2, 0x4, 0x6, 0x8, 0x9, 0xa, 0xc) : (((uint16(@cpio_binary_magic + 6) & 0x3f000) >> 0xc) == i) and
+
+            /**
+                check that c_mode sticky bits is valid
+                C_NONE      000000 (0x0)
+                C_ISUID		004000 (0x4)
+                C_ISGID		002000 (0x2)
+                C_ISVTX		001000 (0x1)
+            */
+            for any j in (0x0, 0x1, 0x2, 0x4): (((uint16(@cpio_binary_magic + 6) & 0xe00) >> 0x9) == j) and
+
+            /**
+                check that c_mode permissions are valid.
+                should be between 0o000 (0x0) (---------) and 0o777 (0x1ff) (rwxrwxrwx)
+            */
+            ((uint16(@cpio_binary_magic + 6) & 0x777) <= 0x1ff)
```

Comment:
It seems like [Yara supports](https://github.com/VirusTotal/yara/blob/1dab22e9f1e9fbb3e5464be19fd0d247ea583065/libyara/lexer.l#L562) octal literals natively. It is funny, that it isn't mentioned at all in the documentation though...

Reply:
Oh wow. Moved to octal notation, which is way easier to understand in this context. I mentioned the link you provided in the commit message so future readers know where this comes from.